### PR TITLE
Display More Informative Error Messages for Bad HTTP Requests and Update "Namespace Get"

### PR DESCRIPTION
**Pull request by dubeejw**
- Display "connection failued, or timed out" when HTTP requests fail
- Update "namespace get" to display the proper namespace when an error occurs.
- Fix defect that prevents a passed namespace from being used with "namespace get"
- Add test to ensure the proper namespace is being used with "namespace get"

Fixes: https://github.com/openwhisk/openwhisk/issues/1203
Fixes: https://github.com/openwhisk/openwhisk/issues/1279
Fixes: https://github.com/openwhisk/openwhisk/issues/1319
PG1: 398

### CODE REVIEW

**Comment by rabbah** [comment]
This improves things:
```
> bin/go-cli/wsk list -i --apihost 192.168.99.100
error: Unable to obtain namespace entities for namespace '': Request failure: The connection failed, or timed out. (code 502)

>  bin/go-cli/wsk action list -i --apihost 192.168.99.100
error: Unable to list action(s): Request failure: The connection failed, or timed out. (code 502)
```

but I find the use of `code` here to be questionable - in that in other cases where we report a code, it means a transaction id (i concede code is not aptly named in that case) but here code is the http status code and this is not consistent.

some other comments: can we reformat the message to avoid the double occurrence of `:`?
also notice the `''` namespace in the first case (should be `default` perhaps).

**Comment by rabbah** [comment]
I noticed the go CLI uses `_` instead of `default` which is more descriptive.

```
> wsk -i list
entities in namespace: default
> bin/go-cli/wsk -i list
Entities in namespace: _
```

File: tests/src/system/basic/WskBasicTests.scala [hunk 1]
```diff
@@ -317,4 +318,11 @@ class WskBasicTests
         wsk.namespace.get(expectedExitCode = SUCCESS_EXIT)(WskProps()).
             stdout should include("default")
     }
+
+    it should "not list entities with an invalid namespace" in {
+        val namespace = "fakeNamespace"
+        val stderr = wsk.namespace.get(Some(s"/${namespace}"), expectedExitCode = ANY_ERROR_EXIT).stderr
```
Comment (rabbah) [inline tests/src/system/basic/WskBasicTests.scala]: the exist code should be not authorized.

Comment (rabbah) [inline tests/src/system/basic/WskBasicTests.scala]: do we have a test that does `wsk namespace get /guest`?

Comment (dubeejw) [inline tests/src/system/basic/WskBasicTests.scala]: Python and Go CLIs both return an exit code of FORBIDDEN (147).

**Comment by rabbah** [comment]
good for the case where controller is up but `_` appears for the failed case where no controller is up:

```
> ./bin/go-cli/wsk -i list
error: Unable to obtain namespace entities for namespace '_': The connection failed, or timed out. (HTTP status code 502)
```

**Comment by dubeejw** [comment]
Anything else on this one?

**Comment by dubeejw** [comment]
PG2 206.

**Comment by rabbah** [comment]
```
> wsk list 
error: Unable to obtain namespace entities for namespace 'default': The connection failed, or timed out. (HTTP status code 502)
> wsk action list
error: Unable to list action(s): The connection failed, or timed out. (HTTP status code 502)
> wsk trigger list
error: Unable to obtain the trigger list for namespace '_': The connection failed, or timed out. (HTTP status code 502)
> wsk rule list
error: Unable to obtain list of rules: The connection failed, or timed out. (HTTP status code 502)
> wsk activation list
error: Unable to obtain list of activations: The connection failed, or timed out. (HTTP status code 502)
> wsk package list 
error: Unable to obtain package list: The connection failed, or timed out. (HTTP status code 502)
```

almost - the error messages should be more consistent and there's one more instance of `_`.

File: tools/cli/go-whisk-cli/commands/action.go [hunk 2]
```diff
@@ -357,9 +357,8 @@ var actionListCmd = &cobra.Command{
     actions, _, err := client.Actions.List(qName.entityName, options)
     if err != nil {
       whisk.Debug(whisk.DbgError, "client.Actions.List(%s, %#v) error: %s\n", qName.entityName, options, err)
-      errMsg := fmt.Sprintf(
-        wski18n.T("Unable to list action(s): {{.err}}",
-          map[string]interface{}{"err": err}))
+      errMsg := wski18n.T("Unable to obtain the action list for namespace '{{.name}}': {{.err}}",
```
Comment (rabbah) [inline tools/cli/go-whisk-cli/commands/action.go]: Unable to obtain the list of actions for namespace ...

File: tools/cli/go-whisk-cli/commands/activation.go [hunk 3]
```diff
@@ -91,10 +91,10 @@ var activationListCmd = &cobra.Command{
         activations, _, err := client.Activations.List(options)
         if err != nil {
             whisk.Debug(whisk.DbgError, "client.Activations.List() error: %s\n", err)
-            errStr := fmt.Sprintf(
-                wski18n.T("Unable to obtain list of activations: {{.err}}",
-                    map[string]interface{}{"err": err}))
-            werr := whisk.MakeWskErrorFromWskError(errors.New(errStr), err, whisk.EXITCODE_ERR_GENERAL, whisk.DISPLAY_MSG, whisk.NO_DISPLAY_USAGE)
+            errStr := wski18n.T("Unable to obtain the activation list for namespace '{{.name}}': {{.err}}",
```
Comment (rabbah) [inline tools/cli/go-whisk-cli/commands/activation.go]: the list of activations

File: tools/cli/go-whisk-cli/commands/namespace.go [hunk 4]
```diff
@@ -90,24 +90,18 @@ var namespaceGetCmd = &cobra.Command{
...
         namespace, _, err := client.Namespaces.Get(qName.namespace)
+
         if err != nil {
-            whisk.Debug(whisk.DbgError, "client.Namespaces.Get(%s) error: %s\n", qName.namespace, err)
-            errStr := fmt.Sprintf(
-                wski18n.T("Unable to obtain namespace entities for namespace '{{.namespace}}': {{.err}}",
-                    map[string]interface{}{"namespace": qName.namespace, "err":err}))
-            werr := whisk.MakeWskErrorFromWskError(errors.New(errStr), err, whisk.EXITCODE_ERR_NETWORK, whisk.DISPLAY_MSG, whisk.NO_DISPLAY_USAGE)
+            whisk.Debug(whisk.DbgError, "client.Namespaces.Get(%s) error: %s\n", getClientNamespace(), err)
+            errStr := wski18n.T("Unable to obtain namespace entities for namespace '{{.namespace}}': {{.err}}",
```
Comment (rabbah) [inline tools/cli/go-whisk-cli/commands/namespace.go]: Unable to obtain the list entities for namespace ...

File: tools/cli/go-whisk-cli/commands/package.go [hunk 5]
```diff
@@ -458,8 +458,8 @@ var packageListCmd = &cobra.Command{
     packages, _, err := client.Packages.List(options)
     if err != nil {
       whisk.Debug(whisk.DbgError, "client.Packages.List(%+v) failed: %s\n", options, err)
-      errStr := fmt.Sprintf(
-        wski18n.T("Unable to obtain package list: {{.err}}", map[string]interface{}{"err":err}))
+      errStr := wski18n.T("Unable to obtain the package list for namespace '{{.name}}': {{.err}}",
```
Comment (rabbah) [inline tools/cli/go-whisk-cli/commands/package.go]: the list of packages ...

File: tools/cli/go-whisk-cli/commands/rule.go [hunk 6]
```diff
@@ -480,9 +480,8 @@ var ruleListCmd = &cobra.Command{
         rules, _, err := client.Rules.List(ruleListOptions)
         if err != nil {
             whisk.Debug(whisk.DbgError, "client.Rules.List(%#v) error: %s\n", ruleListOptions, err)
-            errStr := fmt.Sprintf(
-                wski18n.T("Unable to obtain list of rules: {{.err}}",
-                    map[string]interface{}{"err": err}))
+            errStr := wski18n.T("Unable to obtain the rule list for namespace '{{.name}}': {{.err}}",
```
Comment (rabbah) [inline tools/cli/go-whisk-cli/commands/rule.go]: the list of rules ...

**Comment by rabbah** [comment]
Apologies for one last round of nits.

File: tools/cli/go-whisk-cli/commands/namespace.go [hunk 7]
```diff
@@ -90,24 +90,18 @@ var namespaceGetCmd = &cobra.Command{
...
         namespace, _, err := client.Namespaces.Get(qName.namespace)
+
         if err != nil {
-            whisk.Debug(whisk.DbgError, "client.Namespaces.Get(%s) error: %s\n", qName.namespace, err)
-            errStr := fmt.Sprintf(
-                wski18n.T("Unable to obtain namespace entities for namespace '{{.namespace}}': {{.err}}",
-                    map[string]interface{}{"namespace": qName.namespace, "err":err}))
-            werr := whisk.MakeWskErrorFromWskError(errors.New(errStr), err, whisk.EXITCODE_ERR_NETWORK, whisk.DISPLAY_MSG, whisk.NO_DISPLAY_USAGE)
+            whisk.Debug(whisk.DbgError, "client.Namespaces.Get(%s) error: %s\n", getClientNamespace(), err)
+            errStr := wski18n.T("Unable to obtain the list entities for namespace '{{.namespace}}': {{.err}}",
```
Comment (rabbah) [inline tools/cli/go-whisk-cli/commands/namespace.go]: missing 'of' after list

File: tools/cli/go-whisk-cli/wski18n/resources/en_US.all.json [hunk 8]
```diff
@@ -84,12 +84,12 @@
     "translation": "'{{.name}}' is not a valid qualified name: {{.err}}"
   },
   {
-    "id": "Unable to obtain namespace entities for namespace '{{.namespace}}': {{.err}}",
-    "translation": "Unable to obtain namespace entities for namespace '{{.namespace}}': {{.err}}"
+    "id": "Unable to obtain the list entities for namespace '{{.namespace}}': {{.err}}",
```
Comment (rabbah) [inline tools/cli/go-whisk-cli/wski18n/resources/en_US.all.json]: list of

File: tools/cli/go-whisk-cli/wski18n/resources/en_US.all.json [hunk 9]
```diff
@@ -84,12 +84,12 @@
...
+    "translation": "Unable to obtain the list entities for namespace '{{.namespace}}': {{.err}}"
```
Comment (rabbah) [inline tools/cli/go-whisk-cli/wski18n/resources/en_US.all.json]: the list of

**Comment by rabbah** [comment]
Merged as https://github.com/openwhisk/openwhisk/commit/875d4f53365c1e60ec958e71e11b1f5183ff9071.

**Pull request closed without merging**